lightning/src/ln/channelmonitor.rs

```diff
@@ -1772,11 +1772,11 @@ impl ChannelMonitor {
 					let mut inputs_info = Vec::new();
 
 					macro_rules! sign_input {
-						($sighash_parts: expr, $input: expr, $amount: expr, $preimage: expr) => {
+						($sighash_parts: expr, $input: expr, $amount: expr, $preimage: expr, $idx: expr) => {
```

Comment:
If you're gonna stop relying on the stupid hack of shoving the idx in the sequence field (which is a good cleanup), can you do it in its own commit and fully remove it instead of only partially?

Reply:
That was a quick fix, given than in #462, the macro and a bunch of codes around is going away. Split in its own commit anyway (bundled there because of the very basic dedup parser in test_utils relying on nSequence)

Reply:
Right, but its easier to review if you just do it all the way...this patch seems to pass tests:

```
diff --git a/lightning/src/ln/channelmonitor.rs b/lightning/src/ln/channelmonitor.rs
index d55db634..8a393be9 100644
--- a/lightning/src/ln/channelmonitor.rs
+++ b/lightning/src/ln/channelmonitor.rs
@@ -1805,17 +1805,16 @@ impl ChannelMonitor {
                                                        }
                                                        if let Some(payment_preimage) = self.payment_preimages.get(&htlc.payment_hash) {
                                                                if htlc.offered {
-                                                                       let mut input = TxIn {
+                                                                       let input = TxIn {
                                                                                previous_output: BitcoinOutPoint {
                                                                                        txid: commitment_txid,
                                                                                        vout: transaction_output_index,
                                                                                },
                                                                                script_sig: Script::new(),
-                                                                               sequence: idx as u32, // reset to 0xfffffffd before sign_input
+                                                                               sequence: 0xff_ff_ff_fd,
                                                                                witness: Vec::new(),
                                                                        };
                                                                        if htlc.cltv_expiry > height + CLTV_SHARED_CLAIM_BUFFER {
-                                                                               input.sequence = 0xff_ff_ff_fd;
                                                                                inputs.push(input);
                                                                                inputs_desc.push(if htlc.offered { InputDescriptors::OfferedHTLC } else { InputDescriptors::ReceivedHTLC });
                                                                                inputs_info.push((payment_preimage, tx.output[transaction_output_index as usize].value, htlc.cltv_expiry, idx));
@@ -1834,8 +1833,6 @@ impl ChannelMonitor {
                                                                                let height_timer = Self::get_height_timer(height, htlc.cltv_expiry);
                                                                                let mut used_feerate;
                                                                                if subtract_high_prio_fee!(self, fee_estimator, single_htlc_tx.output[0].value, predicted_weight, used_feerate) {
-                                                                                       let idx = single_htlc_tx.input[0].sequence;
-                                                                                       single_htlc_tx.input[0].sequence = 0xff_ff_ff_fd;
                                                                                        let sighash_parts = bip143::SighashComponents::new(&single_htlc_tx);
                                                                                        let (redeemscript, htlc_key) = sign_input!(sighash_parts, single_htlc_tx.input[0], htlc.amount_msat / 1000, payment_preimage.0.to_vec(), idx);
                                                                                        assert!(predicted_weight >= single_htlc_tx.get_weight());
@@ -1868,7 +1865,7 @@ impl ChannelMonitor {
                                                                                vout: transaction_output_index,
                                                                        },
                                                                        script_sig: Script::new(),
-                                                                       sequence: idx as u32,
+                                                                       sequence: 0xff_ff_ff_fd,
                                                                        witness: Vec::new(),
                                                                };
                                                                let mut timeout_tx = Transaction {
@@ -1884,8 +1881,6 @@ impl ChannelMonitor {
                                                                let height_timer = Self::get_height_timer(height, htlc.cltv_expiry);
                                                                let mut used_feerate;
                                                                if subtract_high_prio_fee!(self, fee_estimator, timeout_tx.output[0].value, predicted_weight, used_feerate) {
-                                                                       let idx = timeout_tx.input[0].sequence;
-                                                                       timeout_tx.input[0].sequence = 0xff_ff_ff_fd;
                                                                        let sighash_parts = bip143::SighashComponents::new(&timeout_tx);
                                                                        let (redeemscript, htlc_key) = sign_input!(sighash_parts, timeout_tx.input[0], htlc.amount_msat / 1000, vec![0], idx);
                                                                        assert!(predicted_weight >= timeout_tx.get_weight());

```